build.rs

```diff
@@ -65,9 +65,12 @@ fn build_and_link() -> IncludePaths {
 
     // Generate configure, run configure, make, make install
     run_command("Generating configure",
-                Command::new("./autogen.sh").current_dir(&build_dir));
+                Command::new("sh")
+                    .arg("autogen.sh")
+                    .current_dir(&build_dir));
```

Comment:
With these changes, the AppVeyor output is now:

```
error: failed to run custom build command for `libffi-sys v0.6.3-alpha.0 (C:\projects\libffi-sys-rs)`
process didn't exit successfully: `C:\projects\libffi-sys-rs\target\debug\build\libffi-sys-79e2ed955f55c39c\build-script-build` (exit code: 101)
--- stderr
sh: autogen.sh: No such file or directory
thread 'main' panicked at 'Generating configure', build.rs:52:9
```

This is a bit odd, since the `autogen.sh` file _should_ be in place. 

I think AppVeyor might not be cloning the submodules. According to https://github.com/appveyor/ci/issues/899#issuecomment-426850994 we need to add the following to `appveyor.yml`:

```
install:
  - cd %APPVEYOR_BUILD_FOLDER%
  - git submodule update --init --recursive
```

I'll try to add this via this PR, assuming AppVeyor picks that up.

Reply:
```
error: failed to run custom build command for `libffi-sys v0.6.3-alpha.0 (C:\projects\libffi-sys-rs)`
process didn't exit successfully: `C:\projects\libffi-sys-rs\target\debug\build\libffi-sys-79e2ed955f55c39c\build-script-build` (exit code: 101)
--- stderr
autogen.sh: line 2: exec: autoreconf: not found
thread 'main' panicked at 'Generating configure', build.rs:52:9
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

Progress! I'll continue later tonight to make sure all dependencies necessary are installed.

Reply:
@tov So I got libffi-sys-rs to run on AppVeyor, but it now fails with [the following](https://ci.appveyor.com/project/tov/libffi-sys-rs/builds/19475805/job/4ix29ywsqv3bhy1v):

```
autoreconf: Leaving directory `.'
Unescaped left brace in regex is deprecated here (and will be fatal in Perl 5.32), passed through in regex; marked by <-- HERE in m/^\s+@([[:alnum:]][[:alnum:]\-]*)({ <-- HERE })?\s*/ at /usr/share/texinfo/Texinfo/Parser.pm line 5481.
Unescaped left brace in regex is deprecated here (and will be fatal in Perl 5.32), passed through in regex; marked by <-- HERE in m/^\s+@([[:alnum:]][[:alnum:]\-]*)({ <-- HERE })?\s*(\@(c|comment)((\@|\s+).*)?)?/ at /usr/share/texinfo/Texinfo/Parser.pm line 5485.
../src/x86/unix64.S: Assembler messages:
../src/x86/unix64.S:45: Warning: .type pseudo-op used outside of .def/.endef: ignored.
../src/x86/unix64.S:45: Error: junk at end of line, first unrecognized character is `f'
../src/x86/unix64.S:204: Warning: .size pseudo-op used outside of .def/.endef: ignored.
../src/x86/unix64.S:204: Error: junk at end of line, first unrecognized character is `f'
../src/x86/unix64.S:208: Warning: .type pseudo-op used outside of .def/.endef: ignored.
../src/x86/unix64.S:208: Error: junk at end of line, first unrecognized character is `f'
../src/x86/unix64.S:326: Warning: .size pseudo-op used outside of .def/.endef: ignored.
../src/x86/unix64.S:326: Error: junk at end of line, first unrecognized character is `f'
../src/x86/unix64.S:334: Error: junk at end of line, first unrecognized character is `,'
make[2]: *** [Makefile:1399: src/x86/unix64.lo] Error 1
make[1]: *** [Makefile:1657: install-recursive] Error 1
make: *** [Makefile:3236: install] Error 2
thread 'main' panicked at 'Building libffi', build.rs:52:9
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

Do you have any idea what might be causing this?

Reply:
Reading through the libffi README, I think this would be required:

> It's also possible to build libffi on Windows platforms with Microsoft's Visual
> C++ compiler. In this case, use the msvcc.sh wrapper script during configuration
> like so:
> 
>     path/to/configure CC=path/to/msvcc.sh CXX=path/to/msvcc.sh LD=link CPP="cl -nologo -EP"
> 
> For 64-bit Windows builds, use CC="path/to/msvcc.sh -m64" and
> CXX="path/to/msvcc.sh -m64". You may also need to specify --build appropriately.

I think we'd have to incorporate this into the build script somehow.

Reply:
I made a bunch of changes to `build.rs`, I'll keep an eye on the build to see if this resolves the remaining issues.